src/com/soomla/store/billing/google/SoomlaGpVerification.java

```diff
@@ -124,6 +124,11 @@ public void verifyPurchase() {
                             } else {
                                 errorCode = UnexpectedStoreErrorEvent.ErrorCode.VERIFICATION_FAIL;
                                 SoomlaUtils.LogError(TAG, "Failed to verify transaction receipt. The user will not get what he just bought.");
+                                try {
+                                    SoomlaUtils.LogError(TAG, "Reason: " + resultJsonObject.getJSONObject("reason").getJSONObject("error").getString("message"));
```

Comment:
@EugeneButusov, please, make this call safer, checking "reason", "error" for null.